_episodes/38.md

```diff
@@ -0,0 +1,525 @@
+adding or removing columns as part of the function, or it can be as complex as
+building and returning an entirely new table based on some input data.
+
+## Demo of the episode: The many ways you can leverage PTFs
+
+But we've talked enough at a high level about what PTFs are, so now it's a good
+time to look at what PTFs can actually do for you to make your life as a Trino
+user easier, better, and more efficient.
+
+### Possible polymorphic table functions
+
+One thing to note - all the examples we're about to look at are *hypothetical*.
+We're working to bring these to Trino soon, but there's a few things left to
+implement before we get there, so for now, this is why we're implementing PTFs,
+and we'll take a look at what you can currently do with them a little later.
+
+#### Select except
+
+Imagine a table with 10 columns, named col1, col2, col3, etc. If you want to
+select all the columns except the first one from that table, you end up with a
+query that looks like:
+
+```sql
+SELECT
+    col2, col3, col4, col5, col6, col7, col8, col9, col10
+FROM
+    my.table;
+```
+
+But that's long, and it's a pain to type, and it gets messy, especially if your
+column names aren't extremely short due to being part of a contrived example.
+With a simple PTF, you could get the same result with:
+
+```sql
+SELECT
+    *
+FROM
+    TABLE(
+        excl_function(
+            data => TABLE(my.table), columns_to_exclude => DESCRIPTOR("col1")
+        )
+    );
+```
+
+Now, this isn't a great PTF, because it's going to take more time to implement
+than it takes to just write out your column names, and at least when we're using
+only 10 columns and short column names, invoking the function takes more writing
+than doing it the old-fashioned way. Also, this is going to perform worse than
+writing the query the ordinary way. As a rule of thumb, if it can be written
+with normal SQL, it will be more performant when done that way. There are plans
+to work on optimizing PTFs, but that's not going to happen soon, so for the time
+being, we're focusing on how they enable things which previously couldn't
+be done at all, rather than making queries look nicer or cleaner.
+
+All that said, we wanted to include this example because this does a good job at
+demonstrating how polymorphic table functions can work and what they can do for
+you. But it's a simple example, and now we can look at some which are a little
+more complex and a little more practical.
+
+#### CSVreader
+
+If you've ever tried to create a table from a CSV file, you know it can be a
+painful experience. It has to be very explicit, very diligent, and there's a lot
+of manual cross-checking involved in ensuring that each column aligns perfectly
+and is correctly typed for the columns present in the CSV. Enter polymorphic
+table functions, here to save the day.
+
+The table function itself will be defined on the connector, so all the end user
```

Comment:
This seems to imply that 1) we're going to implement this function and 2) it will look exactly as described below. Both of those statements are incorrect. We may never implement this particular function, and if we do, it will likely end up looking very differently.

We should make it clear that these are just some hypothetical examples.

Reply:
I'll update to make it clearer, but we do have a paragraph above (line 170) being very explicit that everything which follows is hypothetical.